src/main/scala/com/localytics/sbt/s3/S3ProxyTasks.scala

```diff
@@ -0,0 +1,72 @@
+package com.localytics.sbt.s3
+
+import java.io.File
+import java.net.URL
+
+import com.localytics.sbt.s3.S3ProxyKeys._
+import com.localytics.sbt.s3.S3ProxyUtils._
+import sbt.Keys._
+import sbt._
+
+object S3ProxyTasks {
+
+  def downloadS3ProxyTask = (s3ProxyVersion, s3ProxyDownloadUrl, s3ProxyDownloadDir, s3ProxyDownloadFile, streams) map {
+    case (ver, url, downloadDir, downloadFile, streamz) =>
+      import sys.process._
+      val outputFile = new File(downloadDir, downloadFile)
+      if (!downloadDir.exists()) {
+        streamz.log.info(s"Creating S3Proxy directory $downloadDir")
+        downloadDir.mkdirs()
+      }
+      if (!outputFile.exists()) {
+        streamz.log.info(s"Downloading S3Proxy from [$url] to [${outputFile.getAbsolutePath}]")
+        (new URL(url) #> outputFile).!!
+      }
+      if (!outputFile.exists()) {
+        sys.error(s"Cannot to find S3Proxy file at [${outputFile.getAbsolutePath}]")
+      }
+  }
+
+  def startS3ProxyTask = (downloadS3Proxy, s3ProxyDownloadDir, s3ProxyDownloadFile, s3ProxyPort, s3ProxyDataDir, streams) map {
+    case (_, downloadDir, downloadFile, port, dataDir, streamz) =>
+      val args = Seq("java") ++
+        Seq("-Djclouds.provider=filesystem") ++
+        Seq("-Djclouds.identity=identity") ++
+        Seq("-Djclouds.credential=credential") ++
+        Seq("-Ds3proxy.authorization=none") ++
+        Seq(s"-Djclouds.filesystem.basedir=$dataDir") ++
+        Seq(s"-Ds3proxy.endpoint=http://127.0.0.1:$port") ++
+        Seq(s"-Ds3proxy.secure-endpoint=https://127.0.0.1:$port") ++
+        Seq("-jar", new File(downloadDir, downloadFile).getAbsolutePath) ++
+        Seq("--properties", "/dev/null")
+
+      if (isS3ProxyRunning(port)) {
+        streamz.log.warn(s"S3Proxy is already running on port $port")
+      } else {
+        streamz.log.info("Starting S3Proxy:")
+        Process(args).run()
+        streamz.log.info("Waiting for S3Proxy")
+        waitForS3Proxy(port, (s: String) => streamz.log.info(s))
+      }
+      if (extractS3ProxyPid("jps".!!).isEmpty) {
+        sys.error(s"Cannot find S3Proxy PID")
+      }
+  }
+
+  def stopS3ProxyTask = (streams, s3ProxyDataDir, s3ProxyCleanAfterStop) map {
+    case (streamz, dataDir, clean) =>
+      extractS3ProxyPid("jps".!!) match {
```

Comment:
Do you need to use `jps -l` here? When I run `jps` on my machine I don't get the name of the jar.

```
~/src/local-backend/$ jps
83826 jar
83928 Jps
78462 MainGenericRunner
```
vs
```
~/src/local-backend/$ jps -l
84448 sun.tools.jps.Jps
83826 elasticmq-server-0.8.12.jar
78462 scala.tools.nsc.MainGenericRunner
```